config/routes.rb

```diff
@@ -1,5 +1,8 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :cinema_halls
+  resources :movies
+  resources :reservations
+  resources :screenings
```

Comment:
You not implementing all available methods, so would be better to explicitly tell which methods are implemented in controllers. I think that you can add for each one resource that kind of code:
```resources :some, only: %i[index show update create destroy]```

Reply:
You're right, thank you. I've changed this.